src/analysis/normed_space/conformal.lean

```diff
@@ -0,0 +1,235 @@
+end
+
+end is_conformal_map
+
+/-- A map `f` is said to be conformal if it has a conformal differential `f'`. -/
+def conformal_at {X Y : Type*}
+  [normed_group X] [normed_group Y] [normed_space ℝ X] [normed_space ℝ Y]
+  (f : X → Y) (x : X) :=
+∃ (f' : X →L[ℝ] Y), has_fderiv_at f f' x ∧ is_conformal_map ℝ f'
+
+lemma conformal_at_id {X : Type*}
+  [normed_group X] [normed_space ℝ X] (x : X) : conformal_at id x :=
+⟨id ℝ X, has_fderiv_at_id _, 1, one_ne_zero, id, by ext; simp⟩
+
+lemma conformal_at_const_smul {X : Type*}
+  [normed_group X] [normed_space ℝ X] {c : ℝ} (h : c ≠ 0) (x : X) :
+  conformal_at (λ (x': X), c • x') x :=
+⟨c • continuous_linear_map.id ℝ X, has_fderiv_at.const_smul (has_fderiv_at_id x) c,
+  c, h, id, by ext; simp⟩
+
+namespace conformal_at
+
+lemma differentiable_at {f : X → Y} {x : X} (h : conformal_at f x) :
+  differentiable_at ℝ f x :=
+let ⟨_, h₁, _, _, _, _⟩ := h in h₁.differentiable_at
+
+lemma comp {f : X → Y} {g : Y → Z} (x : X)
+  (hg : conformal_at g (f x)) (hf : conformal_at f x)  : conformal_at (g ∘ f) x :=
+begin
+  rcases hf with ⟨f', hf₁, cf, hcf, lif, hf₂⟩,
+  rcases hg with ⟨g', hg₁, cg, hcg, lig, hg₂⟩,
+  exact ⟨g'.comp f', has_fderiv_at.comp x hg₁ hf₁, cf.comp cg⟩,
+end
+
+lemma const_smul {f : X → Y} {x : X} {c : ℝ} (hc : c ≠ 0) (hf : conformal_at f x) :
+  conformal_at (c • f) x :=
+(conformal_at_const_smul hc $ f x).comp x hf
+
+/-- A real differentiable map `f` is conformal at point `x` if and only if
+    its differential `f'` at that point scales any inner product by a positive scalar. -/
+lemma _root_.conformal_at_iff {f : E → F} {x : E} {f' : E →L[ℝ] F}
+  (h : has_fderiv_at f f' x) : conformal_at f x ↔ ∃ (c : ℝ) (hc : 0 < c),
+  ∀ (u v : E), ⟪f' u, f' v⟫ = (c : ℝ) * ⟪u, v⟫ :=
+begin
+  split,
+  { rintros ⟨f₁, hf, c₁, hc₁, li, hf₁⟩,
+    exact (hf.unique h) ▸ (is_conformal_map_iff f₁).mp ⟨c₁, hc₁, li, hf₁⟩, },
+  { rintros ⟨c, hc, huv⟩,
+    exact ⟨f', h, (is_conformal_map_iff f').mpr ⟨c, hc, huv⟩⟩, },
+end
+
+/-- The conformal factor of a conformal map at some point `x`. Some authors refer to this function
+    as the characteristic function of the conformal map. -/
+def conformal_factor_at {f : E → F} (x : E) {f' : E →L[ℝ] F}
+  (h : has_fderiv_at f f' x) (H : conformal_at f x) : ℝ :=
+by choose c hc huv using (conformal_at_iff h).mp H; exact c
+
+/-- If a real differentiable map `f` is conformal at a point `x`,
+    then it preserves the angles at that point. -/
+lemma _root_.preserves_angle {f : E → F} {x : E} {f' : E →L[ℝ] F}
+  (h : has_fderiv_at f f' x) (H : conformal_at f x) (u v : E) :
+  inner_product_geometry.angle (f' u) (f' v) = inner_product_geometry.angle u v :=
+let ⟨f₁, h₁, c⟩ := H in h₁.unique h ▸ c.preserves_angle f₁ u v
+
+end conformal_at
+
+
+/-- A map `f` is conformal on a set `s` if it's conformal at every point in that set. -/
+def conformal_on (f : X → Y) (s : set X) :=
+∀ (x : X), x ∈ s → conformal_at f x
```

Comment:
`conformal_on` would actually mean something different -- `∀ (x : X), x ∈ s → conformal_within_at f s x`, where `conformal_within_at` is the predicate 
```
∃ (f' : X →L[ℝ] Y), has_fderiv_within_at f f' s x ∧ is_conformal_map ℝ f'
```
The distinction would matter if, say, you wanted to develop the theory of Riemann surfaces with boundary.

Another way to think about it -- `conformal_on` is supposed to care only about the values of the function on `s`.  If `s` is open, the two definitions imply each other; if it's not they don't.

I think it's perfectly fine not to develop the theory of `conformal_within_at` and `conformal_on` at this stage, but let's just keep the naming convention in line with the rest of the library.  So don't make the a definition `conformal_on`; instead just define the groupoid in terms of `conformal_at`,
```lean
property := λ f u, ∀ x ∈ u, conformal_at f x
```

Reply:
Great!